nifi-nar-bundles/nifi-standard-services/nifi-record-sink-api/src/main/java/org/apache/nifi/record/sink/RetryableIOException.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.record.sink;
+
+import java.io.IOException;
+
+/**
+ * This is a marker class for IOExceptions that may succeed if the operation is performed again, it is a hint to retry the operation
+ */
+public class RetryableIOException extends IOException {
```

Comment:
@mattyb149 I see this exception is created and handled but not necessarily used in this PR. Assuming this would be available for future RecordSinkService implementations?  If so recommend just adding a test if possible for that exception branch.

Reply:
Yes, I thought maybe I (or someone else) can look back at existing impls as well, to see if there are some retryable errors that can occur, and start throwing this instead. Will add a unit test (and all constructors of `IOException` to `RetryableIOException`)